File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main (void)
{
/*check input, define variables and functions*/
    float fchg;
    int ichg;
    int greedy(int change);
    do
    {
        printf("Change owed:");
        fchg = get_float();
    }
    while(fchg <= 0);
/*convert input from dollars to cents and then give result via greedy function*/
    ichg = round(fchg * 100);
    printf("%i\n",greedy(ichg));
return 0;
}
/*function to provide coins used*/
int greedy(int change)
{
    int count = 0;
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;
    while(change / quarter >= 1)
    {
        count++;
        change = change - quarter;
    }
    while(change / dime >= 1)
    {
        count++;
        change = change - dime;
    }
    while(change / nickel >= 1)
    {
        count++;
        change = change - nickel;
    }
    while(change / penny >= 1)
    {
        count++;
        change = change - penny;
    }
return count;
}

File: pset6/cash/cash.py
#cash program ported to python 3#
#################################
import cs50

# use mod to get remainder and only divide by that, adding to coin count on the way.
def greedy(change):
    coins = change // 25
    remain = change % 25
    coins += remain // 10
    remain = remain % 10
    coins += remain // 5
    remain = remain % 5
    coins += remain
    return coins

while True:
    # prompt for input and then convert string to int, if not in the range then prompt again
    fchg = cs50.get_float("Change owed: ")
    if fchg > 0:
        break
    print("Please enter a positive number")

# take float input and convert to integer (cents)
ichg = round(fchg * 100)
print(greedy(ichg))

 